File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/PolarTestApi.swift
/// Copyright © 2026 Polar Electro Oy. All rights reserved.

import Foundation

/// Polar SPO2 test API.
public protocol PolarTestApi {
    /// Get SPO2 test data for a given date period.
    ///
    /// - Requires SDK feature(s): `PolarBleSdkFeature.feature_polar_spo2_test_data`
    ///
    /// - Parameters:
    ///   - identifier: The Polar device ID or BT address.
    ///   - fromDate: The starting date of the period to retrieve data from.
    ///   - toDate: The ending date of the period to retrieve data from.
    /// - Returns: A list of `PolarSpo2TestData` for the specified period.
    func getSpo2Test(identifier: String, fromDate: Date, toDate: Date) async throws -> [PolarSpo2TestData]

    /// - Deprecated: Use ``getSpo2Test(identifier:fromDate:toDate:)`` instead.
    @available(*, deprecated, renamed: "getSpo2Test(identifier:fromDate:toDate:)")
    func getSpo2TestData(identifier: String, fromDate: Date, toDate: Date) async throws -> [PolarSpo2TestData]
}

public extension PolarTestApi {
    @available(*, deprecated, renamed: "getSpo2Test(identifier:fromDate:toDate:)")
    func getSpo2TestData(identifier: String, fromDate: Date, toDate: Date) async throws -> [PolarSpo2TestData] {
        return try await getSpo2Test(identifier: identifier, fromDate: fromDate, toDate: toDate)
    }
}

File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/PolarTestApi.kt
// Copyright © 2026 Polar Electro Oy. All rights reserved.
package com.polar.sdk.api

import com.polar.sdk.api.model.PolarSpo2TestData
import java.time.LocalDate

/**
 * Polar SPO2 test API.
 *
 * Requires feature [PolarBleApi.PolarBleSdkFeature.FEATURE_POLAR_FILE_TRANSFER]
 */
interface PolarTestApi {

    /**
     * Get SPO2 test data for a given period.
     *
     * Requires feature [PolarBleApi.PolarBleSdkFeature.FEATURE_POLAR_SPO2_TEST_DATA].
     *
     * @param identifier The Polar device ID or BT address.
     * @param fromDate The starting date of the period to retrieve data from.
     * @param toDate The ending date of the period to retrieve data from.
     * @return A list of [PolarSpo2TestData] representing the SPO2 test data for the specified period.
     */
    suspend fun getSpo2Test(identifier: String, fromDate: LocalDate, toDate: LocalDate): List<PolarSpo2TestData>
}

